src/generalSettings/index.ts

```diff
@@ -0,0 +1,19 @@
+import BaseApi from '../baseApi';
+import responseHandler from '../utils/responseHandlers';
+import RequestParams from '../utils/requestParams';
+import { GeneralSetting } from './types';
+
+export default class GeneralSettings extends BaseApi {
+  public readonly resourceName = 'generalSettings';
+
+  public read(requestParams?: RequestParams<GeneralSetting> | Object): Promise<GeneralSetting[]> {
+    const params = requestParams instanceof RequestParams ? requestParams.getParams() : requestParams;
+    const response = this._get<GeneralSetting>(`${this.getResourceName()}/read`, { params });
+    return responseHandler.requiredList(response);
+  }
+  public readById(id: number, requestParams?: RequestParams<GeneralSetting> | Object): Promise<GeneralSetting> {
```

Comment:
It would make sense to also have readBySettingType here, so you get one general settings based on setting_type, as its more useful as by id.